presto-main/src/main/java/com/facebook/presto/sql/SqlPathElement.java

```diff
@@ -0,0 +1,83 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.facebook.presto.sql;
+
+import com.fasterxml.jackson.annotation.JsonCreator;
+import com.fasterxml.jackson.annotation.JsonProperty;
+
+import java.util.Objects;
+
+import static java.lang.String.format;
+import static java.util.Objects.requireNonNull;
+
+public class SqlPathElement
+{
+    private final String catalogName;
+    private final String schemaName;
+
+    public SqlPathElement(String schemaName)
+    {
+        this(null, schemaName);
+    }
+
+    @JsonCreator
+    public SqlPathElement(
+            @JsonProperty("catalogName") String catalogName,
+            @JsonProperty("schemaName") String schemaName)
+    {
+        this.catalogName = catalogName;
+        this.schemaName = requireNonNull(schemaName, "schemaName is null");
+    }
+
+    @JsonProperty
+    public String getCatalogName()
+    {
+        return catalogName;
+    }
+
+    @JsonProperty
+    public String getSchemaName()
+    {
+        return schemaName;
+    }
+
+    @Override
+    public boolean equals(Object obj)
+    {
+        if (this == obj) {
+            return true;
+        }
+        if ((obj == null) || (getClass() != obj.getClass())) {
+            return false;
+        }
+        SqlPathElement that = (SqlPathElement) obj;
+        return Objects.equals(catalogName, that.catalogName) &&
+                Objects.equals(schemaName, that.schemaName);
+    }
+
+    @Override
+    public int hashCode()
+    {
+        return Objects.hash(catalogName, schemaName);
+    }
+
+    @Override
+    public String toString()
+    {
+        if (catalogName != null) {
```

Comment:
We can simply do `return format("%s.%s", catalogName, schemaName);` (format will handle null properly)

Reply:
My concern with this was having it formatted to "null.<schema>" or something of the sort - is that not an issue?

Reply:
I am fine either way as `toString` will be used during debugging.

Reply:
A similar implementation to this was necessary in the parser, so I think it makes sense to keep it like this for consistency's sake.